### PR TITLE
feat(legacy): disable public radio page and redirect to login

### PR DESCRIPTION
**Pull request by caveman99**
### Description

Many people don't need the public page and use libretime purely for playout management. This adds the ability to have libretime publicly available but only present the login page to the user.

**I have updated the documentation to reflect these changes**:

no, but i will add documentation if this PR is accepted.

### Testing Notes

**What I did:**

Toggle the new ceckbox on the general settings, log out and back in and check behaviour.

note: this may have conflicts with the trim overbooked PR since the toggle sits in the same place. If both are accepted this needs to be formatted nicely :-)

### CODE REVIEW

**Comment by paddatrapper** [comment]
I like this idea a lot. We regularly don't use the embedded site at all and LT is purely presenter/staff focused

File: legacy/application/forms/GeneralPreferences.php [hunk 1]
```diff
@@ -207,6 +207,22 @@ public function init()
...
+        $radioPageDisabled = new Zend_Form_Element_Checkbox('radioPageDisabled');
+        $radioPageDisabled->setDecorators([
+            'ViewHelper',
+            'Errors',
+            'Label',
+        ]);
+        $radioPageDisabledValue = Application_Model_Preference::getRadioPageDisabled();
+        if ($radioPageDisabledValue == '') {
+            $radioPageDisabledValue = false;
+        }
```
Comment (jooola) [inline legacy/application/forms/GeneralPreferences.php:220]: Please move this code the `Application_Model_Preference::getRadioPageDisabled()` function.

File: legacy/application/forms/GeneralPreferences.php [hunk 2]
```diff
@@ -207,6 +207,22 @@ public function init()
...
+        $radioPageDisabled->addDecorator('Label', ['class' => 'enable-tunein']);
+        $radioPageDisabled->setLabel(_('Disable your public Radio Page and redirect to login?'));
```
Comment (jooola) [inline legacy/application/forms/GeneralPreferences.php:222]:
```suggestion
        $radioPageDisabled->setLabel(_('Disable the public radio page and redirect to the login page?'));
```

File: legacy/application/forms/GeneralPreferences.php [hunk 3]
```diff
@@ -207,6 +207,22 @@ public function init()
...
+        $radioPageDisabled->setDecorators([
+            'ViewHelper',
+            'Errors',
+            'Label',
+        ]);
+        $radioPageDisabledValue = Application_Model_Preference::getRadioPageDisabled();
+        if ($radioPageDisabledValue == '') {
+            $radioPageDisabledValue = false;
+        }
+        $radioPageDisabled->addDecorator('Label', ['class' => 'enable-tunein']);
```
Comment (jooola) [inline legacy/application/forms/GeneralPreferences.php:221]: Why is this 'enable-tunein' ?

File: legacy/application/views/scripts/form/preferences_general.phtml [hunk 4]
```diff
@@ -46,6 +46,9 @@
         <?php echo $this->element->getElement('thirdPartyApi')->render() ?>
 
         <?php echo $this->element->getElement('radioPageLoginButton')->renderViewHelper() ?>
-        <?php echo $this->element->getElement('radioPageLoginButton')->renderLabel() ?>
+        <?php echo $this->element->getElement('radioPageLoginButton')->renderLabel() . '<p/>' ?>
```
Comment (jooola) [inline legacy/application/views/scripts/form/preferences_general.phtml:49]: Why is the `. '<p/>'` needed ? Please remove it.

**Review by jooola — CHANGES_REQUESTED** [review]
Thanks this is awesome, only a few nits.

File: legacy/application/forms/GeneralPreferences.php (hunk 3 above)
Comment (caveman99) [reply legacy/application/forms/GeneralPreferences.php:221]: This is Copy/Paste from the style of the already existing 'Hide login link' checkbox. I think the previous committer hijacked it somewhere...

File: legacy/application/views/scripts/form/preferences_general.phtml (hunk 4 above)
Comment (caveman99) [reply legacy/application/views/scripts/form/preferences_general.phtml:49]: without the page break the checkboxes are in the same line in the UI. I didn't want to introduce new styling, replacing the tunein style for such a simple change.

Comment (caveman99) [reply legacy/application/views/scripts/form/preferences_general.phtml:49]: IF we get all the changes in, i will restyle/update the settings UI to eliminate these classes.

**Review by paddatrapper — APPROVED** [review]

**Comment by paddatrapper** [comment]
LGTM. Please update the documentation to reflect this change

File: docs/user-manual/settings.md [hunk 5]
```diff
@@ -21,6 +21,15 @@ time for the convenience of your station staff. You can also set the day of the
 week that you wish to start your station's weekly schedule on, which defaults
 to Sunday.
 
+:::note
+
+The **Station Timezone** setting can not be modified here and is set in the master
+config file. Please consult
+[this admin manual section](../admin-manual/configuration/#general) on how to
+change it.
```
Comment (jooola) [inline docs/user-manual/settings.md:29]:
```suggestion
The **Station Timezone** setting can not be modified on this page. It is set in the [configuration file](../admin-manual/configuration/#general).
```

Somewhat unrelated to this PR, but we'll let it slip :wink: .

File: docs/user-manual/settings.md [hunk 6]
```diff
@@ -60,15 +69,23 @@ wish. (There is more about this feature in the
...
-The **Allowed CORS URLs** is intended to deal with situations where you want a
-remote site with a different domain to access the API. This is relevant when
-there is a reverse proxy server in front of LibreTime. If you are using a
-reverse proxy, the URLs that will be used to access it should be added here.
+:::note
+
+The **Allowed CORS URLs** you can still see in this screenshot was moved to the
+master config file. Please consult
+[this admin manual section](../admin-manual/configuration/#general) on how to
+change it.
```
Comment (jooola) [inline docs/user-manual/settings.md:77]:
```suggestion
The **Allowed CORS URLs** you can still see in this screenshot was moved to the [configuration file](../admin-manual/configuration/#general).
```

File: docs/user-manual/settings.md (hunk 5 above)
Comment (caveman99) [reply docs/user-manual/settings.md]: Sorry, i have a habit to update the page i am working on, if i see stuff is outdated. Can't really help it :-)

**Review by jooola — APPROVED** [review]

**Comment by jooola** [comment]
Tested this locally, it rocks!

File: docs/user-manual/settings.md (hunk 5 above)
Comment (jooola) [reply docs/user-manual/settings.md]: Great that you did!

**Pull request merged**